bin/index.js

```diff
@@ -167,7 +168,7 @@ bundle(args.module, function (err, bundlePath) {
         processor,
         probe('pando:result'),
         pull.through(log),
-        pull.through(function (x) { process.stdout.write(String(x) + '\n') }),
+        pull.through(function (x) { process.stdout.write(zlib.gunzipSync(x) + '\n') }),
```

Comment:
This is where the problem happens. While this should work when pando is used as a commandline tool, this will not work when the processor is used as a library in another application. This is the use case in the test which fails in the continuous integration. I would suggest you decompress between line 454-455 in src/processor.js instead. The design goal for the processor abstraction is to be a drop-in replacement for pull.asyncMap but with the computation being distributed on different cores and machines.

The bin/index.js file should not contain logic that depends on the internals of the processor implementation (and therefore break abstraction), it should only contain the logic necessary to connect a processor with the command-line interface and Unix pipes.

You can run the same test as for the continuous integration on your machine with 'npm test'.

Reply:
Hi Erick,

I have understood very well why we shouldn't implement this logic inside the bin/index.js file. I am just trying to get a better idea of the source code before I send another commit.